cmd/common-main.go

```diff
@@ -66,6 +66,8 @@ func initConfig() {
 func handleCommonCmdArgs(ctx *cli.Context) {
 	// Set configuration directory.
 	{
+		// Root domain name for virtual-host-style requests.
+		globalDomainName = ctx.String("domain")
```

Comment:
I was thinking we don't need a domain name command flag @krishnasrinivas, instead we can have a environment variable for this? like `MINIO_VIRTUAL_HOST_DOMAIN=` ?  and save it in the config file? 

From the code it looks like the server will honor both virtual style and path style even if virtual style is enabled.  Just like how we have `brower: "on"` we can have a `virtualDomain: "minio.io"` entry?